main.tf

```diff
@@ -88,3 +88,12 @@ resource "google_secret_manager_secret_version" "secret-version" {
   secret      = google_secret_manager_secret.secrets[each.value.name].id
   secret_data = each.value.secret_data
 }
+
+resource "google_secret_manager_secret_iam_binding" "binding" {
+  depends_on = [google_secret_manager_secret.secrets]
+  project    = var.project_id
+  for_each   = { for secret in var.secrets : secret.name => secret }
```

Comment:
Would it be possible to `for_each` on `google_secret_manager_secret.secrets`?  That would remove the need for the `depends_on` which is prefferable.

Reply:
Updated it